Extension/src/LanguageServer/client.ts

```diff
@@ -620,6 +644,57 @@ class DefaultClient implements Client {
         this.model.tagParserStatus.Value = notificationBody.status;
     }
 
+    private updateInactiveRegions(params: InactiveRegionParams): void {
+        let renderOptions: vscode.DecorationRenderOptions = {
+            light: { color: "rgba(175,175,175,1.0)" },
+            dark: { color: "rgba(155,155,155,1.0)" }
+        };
+        let decoration: vscode.TextEditorDecorationType = vscode.window.createTextEditorDecorationType(renderOptions);
+
+        // Recycle the active text decorations when we receive a new set of inactive regions
+        let valuePair: DecorationRangesPair = this.inactiveRegionsDecorations.get(params.uri);
+        if (valuePair) {
+            // The language server will send notifications regardless of whether the ranges have changed. Unfortunately we must check for changes ourselves
+            if (!this.isRangesEqual(valuePair.ranges, params.ranges)) {
+                // Disposing of and resetting the decoration will undo previously applied text decorations
+                valuePair.decoration.dispose();
+                valuePair.decoration = decoration;
+
+                // As vscode.TextEditor.setDecorations only applies to visible editors, we must cache the range for when another editor becomes visible
+                valuePair.ranges = params.ranges;
+            }
+        } else { // The entry does not exist. Make a new one
+            let toInsert: DecorationRangesPair = {
+                decoration: decoration,
+                ranges: params.ranges
+            };
+            this.inactiveRegionsDecorations.set(params.uri, toInsert);
+        }
+
+        // Apply the decorations to all *visible* text editors
+        let editors: vscode.TextEditor[] = vscode.window.visibleTextEditors.filter(e => e.document.uri.toString() === params.uri);
+        for (let e of editors) {
+            e.setDecorations(decoration, params.ranges);
+        }
+    }
+
+    // Helper method to compare two ranges for equality
+    private isRangesEqual(r1: vscode.Range[], r2: vscode.Range[]): boolean {
+        if (r1.length !== r2.length) {
+            return false;
+        }
+
+        for (let e1 of r1) {
+            for (let e2 of r2) {
+                if (!e1.isEqual(e2)) {
```

Comment:
This seems wrong.  It will compare r1[0] against r2[1] and fail.  If we are expecting the arrays to have the same ranges in the same order, this should be a single for-loop iterating from 0 to r1.length-1 (or r2.length-1 since they are equal at this point). Then it checks if r1[n] is equal to r2[n].

Reply:
Are the arrays always going to have the Range in the same order?

Reply:
D'oh. This is n squared as well. Thanks